### DISCUSSION TITLE
Do package subpackages support different patches for the same sourcecode tree without cloning the same source again?

### DISCUSSION THREAD

**Discussion by JeffreyO** [termux/termux-packages#17450] [Q&A] [answered]
I'm trying to get APKtool working due to boredom. As a hurdle for that, I need to get AAPT tool patched and built for what the tool expects due to the commits mentioned in this comment: https://github.com/termux/termux-packages/issues/11521#issuecomment-1641130858

I'm currently trying to see if using subpackages would work for this. I want to apply different patches on top of the AAPT tool that is currently maintained by Termux along with the existing patches. Would using subpackages make this possible or will I need to clone the sourcecode again to get a clean base to apply patches to?

**Comment by sylirre** [comment, marked as answer]
Why do you think subpackages can help with that? The purpose of subpackages is only to split original package on multiple parts, e.g. to reduce dependencies weight or reduce package size by splitting optional parts into own package.

Subpackages won't allow using separate patches. That's intended to ensure that all parts were built from the same source code.

Your options:

* Update patches of existing `aapt` package. *My suggestion: consider this variant.*
* Create new package `apktool-aapt` with different patches and make it dependency of `aapt`.
* Build `aapt` within `apktool` package and thus keep patches in `apktool` directory.
